planner/core/resolve_indices.go

```diff
@@ -634,11 +634,44 @@ func (p *baseSchemaProducer) ResolveIndices() (err error) {
 
 // ResolveIndices implements Plan interface.
 func (p *basePhysicalPlan) ResolveIndices() (err error) {
-	for _, child := range p.children {
+	for i, child := range p.children {
 		err = child.ResolveIndices()
 		if err != nil {
 			return err
 		}
+		if p.self != nil {
+			switch p.self.(type) {
+			case *PhysicalHashJoin, *PhysicalMergeJoin, *PhysicalIndexHashJoin, *PhysicalIndexJoin, *PhysicalIndexMergeJoin:
```

Comment:
We don't have to check the type here. I think it's better to call "AlignNullFlag" at "basePhysicalJoin", that's the common embeded type of joins.

Reply:
I mean implement ResolveIndices for "basePhysicalJoin" too

Reply:
done